test/firebase_auth_mocks_test.dart

```diff
@@ -484,6 +492,18 @@ void main() {
     );
   });
 
+  test('User.linkWithCredential can throw exception', () async {
+    final auth = MockFirebaseAuth(signedIn: true, mockUser: tUser);
+    final user = auth.currentUser;
+    final credential =
+        AuthCredential(providerId: 'providerId', signInMethod: 'signInMethod');
+    tUser.exception = FirebaseAuthException(code: 'verification-failure');
+    expect(
+      () async => await user?.linkWithCredential(credential),
```

Comment:
You can safely remove the `async` and `await`, since this is equivalent to writing a method that returns a Future.